spikewrap/pipeline/load_data.py

```diff
@@ -0,0 +1,95 @@
+from ..utils import utils
+
+
+def load_data(
+    base_path: Union[Path, str],
+    sub_name: str,
+    sessions_and_runs: Dict[str, List[str]],
+    data_format: str = "spikeglx",
+) -> PreprocessingData:
+    """
+    Load raw data (in rawdata). If multiple runs are selected
+    in run_names, these will be stored as segments on a SpikeInterface
+    recording object.
+
+    Parameters
+    -----------
+
+    base_path : Union[Path, str]
+        Path where the rawdata folder containing subjects.
+
+    sub_name : str
+        Subject to preprocess. The subject top level dir should reside in
+        base_path/rawdata/ .
+
+    sessions_and_runs : Dict[str, Union[str, List]]
+        A dictionary containing the sessions and runs to run through the pipeline.
+        Each session should be a session-level folder name residing in the passed
+        `sub_name` folder. Each session to run is a key in the
+        `sessions_and_runs` dict.
+        For each session key, the value can be a single run name (str)
+        or a list of run names. The runs will be processed in the
+        order passed.
+
+    data_format : str
+        The data type format to load. Currently only "spikeglx" is accepted.
+
+    Returns
+    -------
+
+    PreprocessingData class containing SpikeInterface recording object and information
+    on the data filepaths.
+
+    TODO
+    ----
+    Figure out the format from the data itself, instead of passing as argument.
+    Do this when adding the next supported format.
+    """
+    empty_data_class = PreprocessingData(Path(base_path), sub_name, sessions_and_runs)
+
+    if data_format == "spikeglx":
+        return _load_spikeglx_data(empty_data_class)
+
+    raise RuntimeError("`data_format` not recognised.")
+
+
+# --------------------------------------------------------------------------------------
+# Format-specific Loaders
+# --------------------------------------------------------------------------------------
+
+
+def _load_spikeglx_data(preprocess_data: PreprocessingData) -> PreprocessingData:
+    """
+    Load raw SpikeGLX data (in rawdata). If multiple runs are selected
+    in run_names, these will be stored as segments on a SpikeInterface
+    recording object.
+
+    See load_data() for parameters.
+    """
+    for ses_name, run_name in preprocess_data.preprocessing_sessions_and_runs():
+        run_path = preprocess_data.get_rawdata_run_path(ses_name, run_name)
+        assert run_name == run_path.name, "TODO"
+
+        with_sync, without_sync = [
+            se.read_spikeglx(
+                folder_path=run_path,
+                stream_id="imec0.ap",
+                all_annotations=True,
+                load_sync_channel=sync,
+            )
+            for sync in [True, False]
+        ]
+        preprocess_data[ses_name][run_name]["0-raw"] = without_sync
+        preprocess_data.sync[ses_name][run_name] = with_sync
+
+        utils.message_user(f"Raw session data was loaded from {run_path}")
```

Comment:
What about using `logging.info(your_message`?

Reply:
This is a good point, for now the logging is quite complex because it needs to be propagated from SpikeInterface. For now I will keep the `utils.message_user()` as it provides some flexibility down the line when it come to implemented the GUI. It is not in this PR but there is some distinction between `spikewrap` controlled print-messages and the print messages `spikeinterface` outputs which are all funneleld to logs. But this definately warrants further discussion on an upcoming logging PR.